### ISSUE TITLE
Kubernetes: Natively support updating version labels

### ISSUE THREAD

**Issue by linki** [renovatebot/renovate#6812]
**What would you like Renovate to be able to do?**

Renovate can already update Kubernetes manifests with new versions of the docker image specified ([docs](https://docs.renovatebot.com/modules/manager/kubernetes/)).

Now that Kubernetes has a recommended label name for the version (`app.kubernetes.io/version`) amonst others ([docs](https://kubernetes.io/docs/concepts/overview/working-with-objects/common-labels/#labels)) Renovate should update it accordingly as well.

We "hacked" it via [using a custom regex manager](https://github.com/linki/ethereum-deploy/commit/88b0f23d22a9741129f22670c3accc6ac21fe6f2) instead of using the Kubernetes manager. The desired effect can be seen [here](https://github.com/linki/ethereum-deploy/pull/10/files). Using the Kubernetes manager it would just bump the image tag and the label would get out of sync.

**Did you already have any implementation ideas?**

It would be nice to support this natively in the Kubernetes manager (probably somewhere [here](https://github.com/renovatebot/renovate/tree/master/lib/manager/kubernetes)) because there it already filters out non-Kubernetes manifests and it seems to be a more structured and stable approach than the crazy regexp which could also support additional requirements in the future better.

**Are there any workarounds or alternative ideas you've tried to avoid needing this feature?**

A [custom regexManager](https://github.com/linki/ethereum-deploy/commit/88b0f23d22a9741129f22670c3accc6ac21fe6f2) also works so far.

**Is this a feature you'd be interested in implementing yourself?**

With some help getting started I can give it a try.

**Comment by rarkins** [comment]
Hi @linki, thanks for this suggestion. Is my understanding correct that the value of the labels should simply match with the image being updated?

I think it should be enough to just update https://github.com/renovatebot/renovate/blob/master/lib/manager/kubernetes/extract.ts and associated fixtures/tests.

Like with the regex manager, you need to make sure that the right depName/lookupName is extracted but then after that the lookup and replace should be done automatically for you.

FYI it would also be OK to rewrite that extract function to use a YAML parser if you need more context than simple line-based parsing/.

**Comment by linki** [comment]
@rarkins Yes, they should simply match. In the future there might be the need to do some sanitization, like removing the "v" prefix or something like that if such a convention arises. But for now the plain value would be sufficient.

Given the already present fixtures I'm also hopeful that it will be an easy change to do.

If you have any quick pointers on how to run it locally and execute the tests that would be greatly appreciated. Only if you have it on your mind. I didn't check the contributing docs yet so I can also figure it out myself later.

**Comment by rarkins** [comment]
Check out https://github.com/renovatebot/renovate/tree/master/docs/development

Approximate steps:
- Fork and clone locally
- After ensuring pre-requisites, verify that `yarn test` works on master
- Create a branch
- Make sure you have a fixture that matches. Either update the existing or add an additional
- Write a test to verify that the extraction catches all the dependencies you need to update
- Enhance the extract function

If you look at either of the helm managers in Renovate you can see use of the YAML parser instead of a line-by-line approach. If you do decide to refactor to use the YAMl parser then please raise that as a separate PR first. i.e. a refactoring before feature addition

**Comment by linki** [comment]
Thanks! That helps me a lot. I'll take look.
